components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java

```diff
@@ -3396,6 +3407,12 @@ private void removeSecretsFromIDP(IdentityProvider identityProvider) {
                 .getProvisioningConnectorConfigs()) {
             removeSecretsFromProperties(provisioningConnectorConfig.getProvisioningProperties());
         }
+
+        for (IdentityProviderProperty idpProperty : identityProvider.getIdpProperties()) {
+            if (idpProperty.getName().startsWith("__secret__")) {
```

Comment:
shall we add a comment here explaining the logic here?

Reply:
are all the secret properties start with the above literals?

Reply:
Yes, in idp properties secrets fields are identified from this prefix and all confidential properties had this prefix attached to the name.